packages/layers/base/app/utils/schemas/court-order-poa.ts

```diff
@@ -0,0 +1,30 @@
+import type { FormCourtOrderPoa } from '#components'
+import { z } from 'zod'
+
+export function getCourtOrderPoaSchema() {
+  const t = useNuxtApp().$i18n.t
+
+  return z.object({
+    hasPoa: z.boolean().optional(),
+    courtOrderNumber: z.union([
+      z.literal(''),
+      z.string()
+        .min(5, t('connect.validation.minChars', { count: 5 }))
+        .max(20, t('connect.validation.maxChars', { count: 20 }))
+    ]).optional()
+  }).superRefine((data, ctx) => {
+    if (data.hasPoa === true) {
+      if (data.courtOrderNumber === undefined || data.courtOrderNumber === '') {
+        ctx.addIssue({
+          code: 'custom',
+          path: ['courtOrderNumber'],
+          message: t('connect.validation.fieldRequired')
+        })
+      }
+    }
+  })
```

Comment:
For my own knowledge, this is zods way of extending/adding more customized validations?

Reply:
Very cool, thank you!